### PR TITLE
Revamp Notif Logic

### PR DESCRIPTION
**Pull request by vcai122**
Refactors notification logic, allow for multiple tokens per user, restructure to support android (very soon), make views cleaner, etc.

For future PR: fix remaining test cases and management commands.

### CODE REVIEW

File: backend/user/models.py
Comment (tuneerroy) [inline backend/user/models.py:23]: would an `is_dev` field not make sense here too? I don't know the Android notif system but would be surprised if they didn't have similar optionality to iOS.

File: backend/user/models.py
Comment (tuneerroy) [inline backend/user/models.py:59]: I assume you'll delete these commented out code once you merge the PR?

File: backend/user/notifications.py
Comment (tuneerroy) [inline backend/user/notifications.py:52]: thoughts on doing:
```
params = (tokens, title, body, category, is_dev, is_shadow)
if delay:
    send_delayed_notifications(*params, delay=delay)
else:
    send_immediate_notifications(*params)
```

File: backend/user/serializers.py
Comment (tuneerroy) [inline backend/user/serializers.py:23]: just a note so we can delete this code before merge PR?

File: backend/user/views.py
Comment (tuneerroy) [inline backend/user/views.py:43]: yoooo ABC lol

File: backend/user/views.py
Comment (tuneerroy) [inline backend/user/views.py:87]: note on the TODO

File: backend/user/views.py
Comment (tuneerroy) [inline backend/user/views.py:121]: just an opinion but probably prettier to extract this into a `get_tokens()` method within the `post` method but I defer to your judgement

**Review by tuneerroy — APPROVED** [review]
Did we want to handle switching users to holding a list of tokens rather than individual tokens in this PR or a later PR? (for ex, user has Android phone + iPad)

File: backend/user/models.py
Comment (vcai122) [reply backend/user/models.py:23]: after talking to android didn't seem like they had

**Comment by vcai122** [comment]
> Did we want to handle switching users to holding a list of tokens rather than individual tokens in this PR or a later PR? (for ex, user has Android phone + iPad)

I think we do right? I changed the user relation on token to be a foreign key.

**Comment by vcai122** [comment]
> Did we want to handle switching users to holding a list of tokens rather than individual tokens in this PR or a later PR? (for ex, user has Android phone + iPad)

I think we do right? I changed the user relation on token to be a foreign key.

**Pull request merged**
